src/reanimated2/hook/useAnimatedKeyboard.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useRef } from 'react';
+import NativeReanimated from '../NativeReanimated';
+import { makeMutable } from '../core';
+import { AnimatedKeyboardInfo } from '../commonTypes';
+
+export function useAnimatedKeyboard(): AnimatedKeyboardInfo {
+  const ref = useRef<AnimatedKeyboardInfo | null>(null);
+  const listenerId = useRef<number>(-1);
+  const isSubscribed = useRef<boolean>(false);
+
+  if (ref.current === null) {
+    const keyboardEventData: AnimatedKeyboardInfo = {
+      isShown: makeMutable(false),
+      isAnimating: makeMutable(false),
+      height: makeMutable(0),
+    };
+    listenerId.current =
+      NativeReanimated.subscribeForKeyboardEvents(keyboardEventData);
+    ref.current = keyboardEventData;
+    isSubscribed.current = true;
+  }
+  useEffect(() => {
+    if (isSubscribed.current === false && ref.current !== null) {
+      // subscribe again after Fast Refresh
+      NativeReanimated.subscribeForKeyboardEvents(ref.current);
```

Comment:
I think that you should unregister the old listener (`listenerId.current`) and then register it again and save the new ID. You can see at useAniamtedSensor, that there is the same case.